### PR TITLE
Use codecov/codecov-action

### PR DESCRIPTION
**Pull request by skirpichev**
FYI: https://github.com/codecov/codecov-python

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -61,8 +64,8 @@ jobs:
           coverage.xml
           build/coverage/html/
     - name: Publish package on PyPI
-      if: matrix.python-version == 3.9 && github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
-      uses: pypa/gh-action-pypi-publish@v1.4.2
```
Comment (cbm755) [inline .github/workflows/test.yml:59]: I don't know this particular GitHub Action enough to know if this is a good change or not, but it does seem unrelated to the commit message.

Comment (cbm755) [reply .github/workflows/test.yml:59]: It looks like the new form here (`@release/v1`) is the current recommendation from upstream so I think it is a good change.  But probably it should be a deliberate change.

**Review by cbm755 — COMMENTED** [review]
I think this looks good, although I have no experience with coverage on GitHub (on GitLab its integrated into the web UI, here it looks like an external service is being used).

Perhaps before it was computing coverage but not uploading the artifacts elsewhere and this change makes it so the artifacts are uploaded to the third-party for viewing.

I cannot speak to whether that is wanted or not but this looks like it does accomplish that.

**Comment by skirpichev** [comment]
FYI: coverage uploaded to https://app.codecov.io/gh/mpmath/mpmath, only status checks aren't enabled (I can't do this without Fredrik).  But sometimes it's useful to have html/xml coverage data (e.g. codecov.io has bugs).

Now pypi things go to a separate commit and no more exclusions for doctests.

**Comment by cbm755** [comment]
Thanks for explanation.  I have now learned what app.codecov.io is.  I don't understand how it was uploading there before this change, but it appears it somehow was.

Thus this is not a change from the project to adopt a new service but rather one it was already using. 

Is that correct?  If so, I have no objections.

(OTOH, if the previous CI was *not* uploading artifacts to codecov.io, then I think this is something that Fredrik would have to approve).

**Comment by cbm755** [comment]
>  that Fredrik would have to approve).

I should say "someone other than me would have to approve".

**Comment by skirpichev** [comment]
> Thus this is not a change from the project to adopt a new service

Nop.  Just little info for debugging code coverage.  (And BTW the coverage html output is more informative than the codecov.io)

**Review by cbm755 — APPROVED** [review]
My concerns addressed, +1 to merge.
